editoast/src/views/test_app.rs

```diff
@@ -0,0 +1,169 @@
+//! Exposes [TestApp] and [TestAppBuilder] to ease the setup of the
+//! test actix server, database connection pool, and different mocking
+//! components.
+
+use std::{ops::Deref, sync::Arc};
+
+use actix_http::Request;
+use actix_web::{
+    body::BoxBody,
+    dev::{Service, ServiceResponse},
+    middleware::NormalizePath,
+    test::init_service,
+    web::{Data, JsonConfig},
+    App, Error,
+};
+use chashmap::CHashMap;
+
+use crate::{
+    client::{MapLayersConfig, PostgresConfig, RedisConfig},
+    core::CoreClient,
+    error::InternalError,
+    infra_cache::InfraCache,
+    map::MapLayers,
+    modelsv2::{database::connection_pool::create_connection_pool, DbConnectionPoolV2},
+    RedisClient,
+};
+
+pub(crate) struct TestAppBuilder {
+    db_pool: Option<DbConnectionPoolV2>,
+    core_client: Option<CoreClient>,
+    db_pool_v1: bool,
+}
+
+impl TestAppBuilder {
+    pub fn new() -> Self {
+        Self {
+            db_pool: None,
+            core_client: None,
+            db_pool_v1: false,
```

Comment:
This pattern looks like a `enum`.

```rust
enum DbPoolVersion {
  V1,
  V2(DbConnectionPoolV2),
}
```

This would avoid all the asserts below I believe.